src/hivpy/hiv_status.py

```diff
@@ -64,12 +62,31 @@ def set_dummy_viral_load(self, population):
         """Dummy function to set viral load until this part of the code has been implemented properly"""
         population.data[col.VIRAL_LOAD_GROUP] = rng.choice(7, population.size)
 
+    def get_infection_prob(self, sex, age, n_partners, stp_age_groups):
+        # Slow example that avoid repeating the iterations over partners three time by putting them as part of 
+        # one for loop, but for loops in python will be slow.
```

Comment:
Yeah, it'd be nice to be able to vectorise this, but having to repeat it for multiple partners makes it tricky... Will try to think of something for a future version.